SU2_CFD/src/fluid/CFluidFlamelet.cpp

```diff
@@ -0,0 +1,245 @@
+  su2double val_prog = val_scalars[I_PROGVAR];
+
+  string name_enth = table_scalar_names[I_ENTH];
+  string name_prog = table_scalar_names[I_PROGVAR];
+
+  /*--- add all quantities and their address to the look up vectors ---*/
+  look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, val_enth);
+
+  /*--- compute Cv from Cp and molar weight of the mixture (ideal gas) ---*/
+  Cv = Cp - UNIVERSAL_GAS_CONSTANT / molar_weight;
+}
+
+/* --- Total enthalpy is the transported variable, but we usually have temperature as a boundary condition,
+       so we do a reverse lookup */
+unsigned long CFluidFlamelet::GetEnthFromTemp(su2double* val_enth, const su2double val_prog, const su2double val_temp,
+                                              su2double initial_value) {
+  string name_prog = table_scalar_names[I_PROGVAR];
+  string name_enth = table_scalar_names[I_ENTH];
+
+  /* convergence criterion for temperature in [K], high accuracy needed for restarts. */
+  su2double delta_temp_final = 0.001;
+  su2double enth_iter = initial_value;
+  su2double delta_enth;
+  su2double delta_temp_iter = 1e10;
+  unsigned long exit_code = 0;
+  vector<string> look_up_tags;
+  vector<su2double*> look_up_data;
+  int counter_limit = 1000;
+
+  int counter = 0;
+  while ((abs(delta_temp_iter) > delta_temp_final) && (counter++ < counter_limit)) {
+    /* look up temperature and heat capacity */
+    look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, enth_iter);
+
+    /*--- calculate delta_temperature ---*/
+    delta_temp_iter = val_temp - Temperature;
+
+    /* calculate delta_enthalpy following dh = cp * dT */
+    delta_enth = Cp * delta_temp_iter;
+
+    /*--- update enthalpy ---*/
+    enth_iter += delta_enth;
+  }
+
+  /*--- set enthalpy value ---*/
+  *val_enth = enth_iter;
+
+  if (counter >= counter_limit) {
+    exit_code = 1;
+  }
+
+  return exit_code;
+}
+
+void CFluidFlamelet::PreprocessLookUp() {
+  /*--- Set lookup names and variables for all relevant lookup processes in the fluid model ---*/
+
+  /*--- Thermodynamic state variables ---*/
+  varnames_TD.resize(7);
+  val_vars_TD.resize(7);
+
+  /*--- The string in varnames_TD as it appears in the LUT file ---*/
+  varnames_TD[0] = "Temperature";
+  val_vars_TD[0] = &Temperature;
+  varnames_TD[1] = "Density";
+  val_vars_TD[1] = &Density;
+  varnames_TD[2] = "Cp";
+  val_vars_TD[2] = &Cp;
+  varnames_TD[3] = "ViscosityDyn";
+  val_vars_TD[3] = &Mu;
+  varnames_TD[4] = "Conductivity";
+  val_vars_TD[4] = &Kt;
+  varnames_TD[5] = "DiffusionCoefficient";
+  val_vars_TD[5] = &mass_diffusivity;
+  varnames_TD[6] = "MolarWeightMix";
+  val_vars_TD[6] = &molar_weight;
+
+  /*--- Source term variables ---*/
+  varnames_Sources.resize(n_table_sources);
+  val_vars_Sources.resize(n_table_sources);
+
+  for (size_t iSource = 0; iSource < n_table_sources; iSource++) {
+    varnames_Sources[iSource] = table_source_names[iSource];
+    val_vars_Sources[iSource] = &table_sources[iSource];
+  }
```

Comment:
Is there an overload of LookUp_XY where you can pass a vector of su2double by reference for the table to write directly into the vector instead of passing a vector of pointers? This is not very efficient.

Reply:
yes.